backend/upload/management/commands/import-existing.py

```diff
@@ -0,0 +1,141 @@
+        except OSError as err:
+            raise CommandError(
+                'Error accessing BaseX database {}: {}'
+                ' - probably this database does not exist.'
+                .format(dbname, str(err))
+            )
+        return basex_db, words, sentences
+
+    def create_component(self, comp: dict):
+        try:
+            slug = comp['slug']
+            title = comp['title']
+            description = comp['description']
+            databases = comp['databases']
+        except KeyError as err:
+            raise CommandError('Key {} missing in component definition.'
+                               .format(err))
+        component = Component(slug=slug, title=title,
+                              description=description)
+        component.treebank = self.treebank
+        component.variant = comp.get('variant', '')
+        component.group = comp.get('group', '')
+        basex_dbs = []
+        nr_words = 0
+        nr_sentences = 0
+        for dbname in databases:
+            basex_db, words, sentences = self.create_database(dbname)
+            nr_words += words
+            nr_sentences += sentences
+            basex_db.component = component
+            basex_dbs.append(basex_db)
+        component.nr_words = nr_words
+        component.nr_sentences = nr_sentences
+        return component, basex_dbs
+
+    def handle(self, *args, **options):
+        if not basex.test_connection():
+            raise CommandError('Cannot connect to BaseX. '
+                               'This command needs BaseX to run.')
+
+        config_file = options['configuration']
+        try:
+            self.configuration = json.loads(open(config_file, 'r').read())
+        except OSError as err:
+            raise CommandError('Cannot open configuration file: {}.'
+                               .format(str(err)))
+        except json.JSONDecodeError as err:
+            raise CommandError('Cannot parse configuration file: {}.'
+                               .format(str(err)))
+
+        self.treebank = Treebank()
+        try:
+            self.treebank.slug = self.configuration['slug']
+            self.treebank.title = self.configuration['title']
+            self.config_components = self.configuration['components']
+        except KeyError as err:
+            raise CommandError('Key {} missing from configuration file.'
+                               .format(err))
+        if Treebank.objects.filter(slug=self.treebank.slug).count() != 0:
+            raise CommandError('Treebank {} already exists.'
+                               .format(self.treebank.slug))
+
+        # Optional treebank-wide fields that are stored in JSON
+        self.treebank.variants = self.configuration.get('variants', '[]')
+        self.treebank.groups = self.configuration.get('groups', '{}')
+
+        component_objs = []
+        db_objs_per_component = []  # A list of BaseXDB objects per component
```

Comment:
I personally dislike having two separate lists where data is linked by index. Maybe make a list of tuples or a dictionary instead?